### PR TITLE
improve(api): Add quote timestamp buffer to suggested-fees route

### PR DESCRIPTION
**Pull request by nicholaspai**
This should be combined with a change to the relayer code [here](https://github.com/across-protocol/relayer-v2/pull/340) that will make relayer ignore any quote times that do not sufficiently lag HEAD

### CODE REVIEW

File: api/_constants.ts [hunk 1]
```diff
@@ -57,6 +57,9 @@ export const relayerFeeCapitalCostConfig = {
   },
 };
 
+// If `timestamp` is not passed into a suggested-fees query, then return the latest mainnet timestamp minus this buffer.
+export const QUOTE_TIMESTAMP_BUFFER = 12 * 60 * 25; // ~25 blocks on mainnet (12s/block), ~= 5 minutes.
```
Comment (mrice32) [inline api/_constants.ts]: nit: naming issue, looks like below you import DEFAULT_QUOTE_TIMESTAMP_BUFFER not QUOTE_TIMESTAMP_BUFFER.

File: api/suggested-fees.ts [hunk 2]
```diff
@@ -81,8 +94,15 @@ const handler = async (
...
-    const blockTag = isString(timestamp) ? latestBlock : BLOCK_TAG_LAG;
+    // to be "latest". If we're not using the block associated with the `timestamp` passed as a
+    // request param, then we should set the lag approx. equal to the # of blocks that the
+    // quote time buffer lags HEAD.
+    // Note: Its ok if the BLOCK_TAG_LAG doesn't exactly correspond to the parsedTimestamp, since we are returning
+    // an indicative fee, not one passed to a smart contract function. The relayer will compute the actual fee for the
+    // quote time.
+    const blockTag = isString(timestamp)
+      ? latestBlock
+      : (-1 * quoteTimeBuffer) / 60 / 12;
```
Comment (mrice32) [inline api/suggested-fees.ts:105]: I think we can actually get rid of this code entirely and just use latestBlock directly (or change the name above of latestBlock to blockTag).

The latest block should already take into account the delay, right?

Thoughts?

File: api/_constants.ts (hunk 1 above)
Comment (mrice32) [reply api/_constants.ts]: Ah, looks like you fixed.

File: api/suggested-fees.ts (hunk 2 above)
Comment (nicholaspai) [reply api/suggested-fees.ts:105]: yeah on second review, you're right

**Review by mrice32 — APPROVED** [review]
LGTM!

**Pull request merged**